backend/src/main/java/com/redhat/cloud/notifications/IllegalArgumentExceptionMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.redhat.cloud.notifications;
+
+import org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.Response;
+
+// This mapper is required to forward the HTTP 400 error message.
+public class IllegalArgumentExceptionMapper implements ResponseExceptionMapper<IllegalArgumentException> {
```

Comment:
@gwenneg this mapping is somehow strange. Shouldn't it be something like toResponse() instead of toThrowable() ?

What is a good way to add this exception mapper? Just add it to all REST endpoint classes?

Reply:
That's somehow a bit ugly. We probably would have to change NotificationService and InternalService signatures so it returns Response and surround the method calls inside with try, catch and mapping as we are doing for BadRequestException